onnxruntime/core/providers/cpu/optional/optional_ops.cc

```diff
@@ -69,18 +70,15 @@ static void CopySequenceTensor(AllocatorPtr alloc,
   }
 
   tgt->SetType(src->DataType());
-
-  std::vector<Tensor> output_tensors;
-  output_tensors.reserve(src->Size());
+  tgt->Reserve(src->Size());
 
   auto in_tensor = src->begin();
   for (; in_tensor != src->end(); ++in_tensor) {
-    Tensor tmp(in_tensor->DataType(), onnxruntime::TensorShape(in_tensor->Shape()), alloc);
-    CopyCpuTensor(&*in_tensor, &tmp);
-    output_tensors.push_back(std::move(tmp));
+    auto& tensor = in_tensor->Get<Tensor>();
+    Tensor tmp(tensor.DataType(), onnxruntime::TensorShape(tensor.Shape()), alloc);
+    CopyCpuTensor(&tensor, &tmp);
```

Comment:
>CopyCpuTensor(&tensor, &tmp); [](http://example.com/codeflow?start=4&length=29)

I was wondering how do you decide if you need to use DataTransferManager()? All of the kernels are actually CPU kernels, but in Sequence ones you use the manager, because you want to use them for DML. I think we need to document that in the comments, so people do not regress.